sky/clouds/gcp.py

```diff
@@ -524,6 +524,63 @@ def get_vcpus_mem_from_instance_type(
         return service_catalog.get_vcpus_mem_from_instance_type(instance_type,
                                                                 clouds='gcp')
 
+    def get_available_reservation_resources(
+        self,
+        instance_type: str,
+        region: str,
+        zone: Optional[str],
+        specific_reservations: List[str],
+    ) -> int:
+        assert zone is not None, 'GCP requires zone to get available reservations.'
+        reservations = self._list_reservations_for_instance_type_in_zone(
```

Comment:
Should we skip the checks when the `specific_reservtions` are not specified? When I am trying to launch a normal GCP machine, it seems taking significant longer time for optimization than before: `sky launch --cloud gcp --cpus 2+`

Reply:
The current implementation also optimize for `"specificReservationRequired": false` reservations which don't require the specific_reservations argument. Other idea, do we want a CLI argument to enable/disable this feature?

Reply:
If we do the above, we would ignore the non-specific reservations which can result in "lower" cost. Even if you don't provide specific-reservations you can still use them.
![image](https://github.com/skypilot-org/skypilot/assets/29891099/30512327-dc3a-4023-9ba0-fe8e3796cd00)

Reply:
I think, with the optimization that you propose below, we cam close that comment.